src/careamics/models/lvae/likelihoods.py

```diff
@@ -355,6 +356,8 @@ def log_likelihood(self, x: torch.Tensor, params: dict[str, torch.Tensor]):
         torch.Tensor
             The log-likelihood tensor. Shape is (B, C, [Z], Y, X).
         """
+        self.data_mean = self.data_mean.to(x.device)
```

Comment:
I guess this should be done once for all inside the `training_step` __init__. Indeed, here it would be instead called at every `training step`.